[libxkbcommon] Update to 1.3.1. Fixes JB#55591

Tomin1 opened the pull request:
Convert to submodule format. Update it to 1.3.1. Update spec. Use
summary and description texts from Fedora.

Add simple build system patch to skip building of utilities.

mlehtima commented on rpm/libxkbcommon.spec:
```diff
@@ -1,58 +1,58 @@
-BuildRequires:  libtool
 BuildRequires:  flex
 BuildRequires:  bison
+BuildRequires:  xkeyboard-config-devel
```
Would using pkgconfig dependency look better "pkgconfig(xkeyboard-config)"?

Tomin1 replied:
Yes, very much so. :+1: 

Then again we could actually skip that dependency as all it seems to need from that is xkb_base path (seems to be /usr/share/X11/xkb). If we trust that's constant for us, then there is no need to look it up and we can set it with build option instead. Probably doesn't matter very much wrt. build time etc.

saukko commented:
Would split the subtree removal to separate commit so that one can actually see what was changd in addition :)

Tomin1 committed:
Convert to submodule format

Updated upstream to the version the same version as the old subdirectory
used to have. It doesn't build because it's missing disabling of xorg
but otherwise the content is identical.

Signed-off-by: Tomi Leppänen <tomi.leppanen@jolla.com>

Tomin1 committed:
[libxkbcommon] Update to 1.3.1. Fixes JB#55591

Update to 1.3.1. Update spec. Use summary and description texts from
Fedora.

Add simple build system patch to skip building of utilities.

Signed-off-by: Tomi Leppänen <tomi.leppanen@jolla.com>

Tomin1 commented:
> Would split the subtree removal to separate commit so that one can actually see what was changd in addition :)

It's now split.

saukko reviewed (approved):
LGTM from packaging point of view, didn't test on device but looks ok in general.
